test/e2e/external_gateways.go

```diff
@@ -28,20 +28,30 @@ const (
 )
 
 var externalContainerNetwork = "kind"
+var externalContainerIPv4 = ""
+var externalContainerIPv6 = ""
 
 func init() {
 	// When the env variable is specified, we use a different docker network for
 	// containers acting as external gateways
-	exNetwork, found := os.LookupEnv("OVN_TEST_EX_GW_NETWORK")
-	if found {
+	if exNetwork, found := os.LookupEnv("OVN_TEST_EX_GW_NETWORK"); found {
 		externalContainerNetwork = exNetwork
 	}
+
+	// When OVN_TEST_EX_GW_NETWORK is set to "host" we need to set the container's IP from outside
+	if exHostIPv4, found := os.LookupEnv("OVN_TEST_EX_GW_IPV4"); found {
+		externalContainerIPv4 = exHostIPv4
+	}
+
+	if exHostIPv6, found := os.LookupEnv("OVN_TEST_EX_GW_IPV6"); found {
+		externalContainerIPv6 = exHostIPv6
+	}
 }
 
 // gatewayTestIPs collects all the addresses required for a external gateway
 // test.
 type gatewayTestIPs struct {
-	gatewayIPs [2]string
+	gatewayIPs []string
 	srcPodIP   string
 	nodeIP     string
```

Comment:
how do we handle dual stack case here we just single IP for srcpodIP and node ?

Reply:
> not sure I understand but a dual-stack cluster can run both types of tests (ipv4 and ipv6), we have a gatewayTestIPs for each family

I was asking about srcPod and nodeIP this is obviously just one IP while for dual-stack cluster those will be slice of 2 like u have for gwIPs ?

Reply:
I'm not sure, but it's out of the scope of this pr if these tests support dual-stack scenarios or not. here I'm just adding support to run the containers outside of the kind network, without (hopefully) breaking the existing behavior.